srv/modules/runners/validate.py

```diff
@@ -752,27 +752,71 @@ def ceph_version(self):
         local = salt.client.LocalClient()
-        contents = local.cmd(search, 'pkg.latest_version', ['ceph'], tgt_type="compound")
+        contents = local.cmd(search, 'pkg.latest_version', ['ceph-common'], tgt_type="compound")
         for minion, version in contents.items():
+            # sometimes, version contains a string value
+            # other times, it contains an empty dict
+            log.debug("VALIDATE ceph_version: minion ->{}<- latest_version version ->{}<-"
+                      .format(minion, version))
             if not version:
-                info = local.cmd(minion, 'pkg.info_installed', ['ceph'])
-                if info and 'version' in info[minion]['ceph']:
-                    version = info[minion]['ceph']['version']
+                info = local.cmd(minion, 'pkg.info_installed', ['ceph-common'])
+                if info and 'version' in info[minion]['ceph-common']:
+                    version = info[minion]['ceph-common']['version']
+                    log.debug("VALIDATE ceph_version: minion ->{}<- info_installed version ->{}<-"
+                              .format(minion, version))
                 else:
-                    self.errors.setdefault('ceph_version', []).append(
-                        "No Ceph version is available for installation in {}".format(minion))
-                    continue
+                    failmsg = ("No Ceph version is available for installation on minion {}"
+                               .format(minion))
+                    if self.in_dev_env:
+                        log.warning('VALIDATE ceph_version: ' + failmsg)
+                    else:
+                        self.errors.setdefault('ceph_version', []).append(failmsg)
+                        continue
 
             colon_idx = version.find(':')
             if colon_idx != -1:
                 version = version[colon_idx+1:]
             dash_idx = version.rfind('-')
             if dash_idx != -1:
                 version = version[:dash_idx]
-            if version < JEWEL_VERSION:
+            log.debug("VALIDATE ceph_version: minion ->{}<- final munged version ->{}<-"
+                      .format(minion, version))
+            assert isinstance(version, str), "version value is not a string"
+
+            # "11.10" < "11.2" in Python terms, but not in terms of
+            # version numbering semantics, so we have to break the version number
+            # down into its integer components and compare those separately.
+            #
+            # We can assume that Ceph version numbers will always begin with X.Y.Z
+            # where X, Y, and Z are integers. Here, we are only interested in X and Y.
+            #
+            # In other words, there must be at least two version components and
+            # both must be convertible into integers.
+
+            try:
+                version_components = version.split('.')
+                int_version_components = [0, 0]
```

Comment:
It is better to use shorter name for local scope variables, and use array comprehensions. Check the one-liner:
```ver = [int(x) for x in version.split('.')```